[23696] Fix updating wp resource values from allowedValues

oliverguenther opened the pull request:
When a form is activated, fields on the work package resource were
updated from the form's embedded schema, more precisely, from the
`allowedValues` when these existed.

However, they were checked by name instead of the href, which does not
exist on most properties.

https://community.openproject.com/work_packages/23696

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://oliverguenther-fix-23696-custom-field-list.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/' target='_blank'>Open stage</a> or <a href='http://oliverguenther-fix-23696-custom-field-list.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com//teatro.log' target='_blank'>view logs</a>.

oliverguenther committed:
[23696] Fix updating wp resource values from allowedValues

When a form is activated, fields on the work package resource were
updated from the form's embedded schema, more precisely, from the
`allowedValues` when these existed.

However, they were checked by name instead of the href, which does not
exist on most properties.

https://community.openproject.com/work_packages/23696

oliverguenther committed:
Extend custom field spec
